### DISCUSSION TITLE
Audio Suddenly stops

### DISCUSSION THREAD

**Discussion by OrgelSpieler** [GrandOrgue/grandorgue#900] [Help] [answered]
I had this problem over two years ago where the sound would suddenly stop even though GO was receiving midi signals and showed a response to commands on the monitor.  When the sound stopped, I would close GO, then re-open and everything would return to normal.  This occurred for a few months before the problem mysteriously went away.  The system has been rock solid for over 18 months.  I recently upgraded to the latest release and the problem has returned.  Sometimes the system will run for twenty to thirty minutes before the sound stops.  At others times, the sound will stop in a matter of minutes.  I don't have to play the instrument for the sound to halt.  I can restart GO, activate a lone stop and verify the presence of sound.  I can walk away and return a while later, press a key and there will be no sound.  On a lark, I pressed the panic button and the sound returned.  I never had tried that before, but that might offer some clue as to what is causing the problem.  I don't know what the panic button resets internally with respect to the code.

I run a Win 7 Xeon platform with 48 GB of memory and RME HDSPe AIO sound card.  The latter has been on the latest driver version for over a year.  The only other software on this PC are some diagnostic programs for room equalization and the software MusicBee.

My assessment would be that the upgrade reset some register which is causing this problem, but I don't know where to go from here.  If there is some diagnostic mode I could activate, I'd be more than willing to help track this down, otherwise I somewhat at a loss as to what to do.

**Comment by oleg68** [comment]
1. What version of GO are you using exactly?
2. Could you post here the screenshot of your File->Settings->Audio Output?
3. Does the log messge window bring when the sound stops or when you press Panic? If so, please post here it's content.
4. Do you remember, whether you had upgraded the version of GO before the problem went away two years ago?

**Comment by OrgelSpieler** [comment]
Thank you for responding so quickly.  Here are answers to your questions: 

1.  I am running version 3.4.2-1.  I had upgraded to version 3.4.2-1 from  0.3.1.2330-156.3.

2.  [GO Audio Output.pdf](https://github.com/GrandOrgue/grandorgue/files/7684705/GO.Audio.Output.pdf)

3.  No log message window occurs.  When I press panic, there is a quick flash of the screen as if it is being refreshed, but there is no change to anything that appears on the screen.

4  I had not upgraded when the problem went away.  I was on  version 0.3.1.2330-156.3 when the problem occurred and when the problem went away.

**Comment by OrgelSpieler** [comment]
In addition to my last response, I normally have the buffer set to 128 or 256.  I changed it to 512 to see if it made any difference.  It did not.  Using the Midi Events panel, I have verified that there is no midi traffic when the organ is not being played.

**Comment by oleg68** [comment]
1. Are you using several audio cards? 
2. Could you try to use a newer version 3.4.4-1 ?
3. Could you try to switch the audio driver from WDM-KS to some another, ex. ASIO?
4. Could you try to switch the sound port from PA to RT?

**Reply by OrgelSpieler** [reply]
The upgrade to version 3.4.4.-1 did not make a difference.  Her are my answers to your questions:

1.  There are two audio cards in the PC. 
     -  A RME HDSP 9632 is used exclusively for Music Bee
     -  A RME HDSP 9632e AIO is used exclusively for GO - info at: [https://rme-audio.de/hdspe-aio.html]
     
     The interfaces shown in the GO audio output are all present on the same card 9632e AIO card.
     It supports simultaneous operation of Analog, SPDIF,  ADAT and headphone outputs

     The two cards have been running in the same PC for over 15 months without any stability issues

2.  I upgraded to version 3.4.4-1 without making any other changes.  The sound still failed in a matter of minutes.
     The upgrade produced the following error on startup but it did not seem to affect any operations.

      __12/9/2021 11:38:55 AM: 11:38:55 AM: Warning: Index 'C:\Users\OrgelSpieler\AppData\Roaming\GrandOrgueCache     \6CB15696419D4E412B4CCCFDB85041F427AF630F.idx' has bad magic - bypassing index__

3.  I am confused about the support for ASIO drivers.  My sound card supports them, but it is my understanding that ASIO support was removed from GO standard releases a few revisions ago.   Whenever I've seen an inquiry about ASIO in a discussion group, it is always accompanied by the directive to use the standard Windows sound driver WDM-KS or WASPI.  So I'm a bit confused as to the options I have.  How would I install ASIO drivers?

4.  I do not understand the sound port settings.  That is something that was not an option in my 0.3.1.2330-156.3 version.   I'm not sure what I should check and uncheck, so I left the default selections in place.  Can you advise how I might set these or point me to a note that explains how to configure this setting?

Thank you

**Reply by OrgelSpieler** [reply]
I think I have figured out the sound port settings. As I understand it, RT and PA are different kinds of cross-platform audio interfaces.  RT is based on C++ and PA is based on C.  Do I have that right?  Based on my limited knowledge, it's not obvious why I would choose one over the other, but I see that if I select RT, I'm limited to ASIO or Direct Sound Drivers. 

I attempted to change from the (PA)  WDM-KS interface to the (PA) WASAPI but was unsuccessful.  When I click okay, the system gives produces an error "invalid number of channels."  I don't know why that occurs, but if I recall correctly, I got the same error when attempting to use WASAPI in my first usage of GO three years ago.  WDM-KS was the only thing that worked.

**Comment by oleg68** [comment, marked as answer]
There are two risky things

1. Using several audio interfaces at once
2. Using a quite old windows 7

There was a similar problem #702 caused by a bad multithreading synchronisation. It has been fixed by implementings this with the C++ standard objects in #817.

Maybe these standard objects work wrong in Windows 7, so it might not resolve the issue.

For troubleshooting your issues I suggest you to make some experiments.

1. First of all make a backup copy of  the GrandOrgueConfig file that contains your settings
2. Try to remove all audio devices except one. Will the sound stop in this case?
3. Try to install ASIO drivers and to use rt:asio or pa:asio. Will the sound stop in this case?

**Reply by OrgelSpieler** [reply]
Removal of the extra sound card did not make any difference.  When it failed the first time (over 18 mo ago) I was only using the RME HDSP 9632 card at the time  I upgraded to the AIO card about a year ago and kept the other sound card for the Music Bee application.   I was concerned about adding the second card, but Microsoft techs told me that Win7 could support more than one sound card.  My experience did not suggest otherwise as the performance has been flawless.

I don't quite understand how to use an AIO driver.
1.  Do I need a specific build of GO for an ASIO driver to work?  If so, where do I find that?
2.  If I need to obtain an ASIO driver, what is a good source for that?   In what folder should it be installed?

Do you have any ideas on why the WASAPI drivers do not work (the system complains about the number of channels)?

When I first started working with GO in 2017, I didn't see a lot of positive comments in the forums about users working with Windows 10, so I decided to start with Windows 7.  I  haven't seen any reason to change as things have been so stable.  I know that Win7 is EOL with Microsoft, but I don't see that as much of an issue with the way I use it.   I have no axe to grind with Win10, as I run it on other computers.  For this particular application, Win7 seemed like a better choice at the time.  Is my reasoning in error?

**Reply by oleg68** [reply]
> * Do I need a specific build of GO for an ASIO driver to work?  If so, where do I find that?

No, you don't. Your current build 3.4.4-1 is fully compatible with ASIO.

Could you click on the `Change` in the Audio Output sessings and post here the screenshot of available audio devices?

> * If I need to obtain an ASIO driver, what is a good source for that?  

https://rme-audio.de/support.html. Find there your audiocard and download the drivers. But I think you have already installed ASIO drivers. So give the screenshoot from the previous step.

> Do you have any ideas on why the WASAPI drivers do not work (the system complains about the number of channels)?

Because WASAPI disposes audio channels in a different layout than KS. You should assign the channels from scratch. But I recommend you to start with ASIO and two channels then to add more channels. The best layout for GrandOrgue is that the audio driver whould expose all output capabilities as different channels of a single audio interface.

> I haven't seen any reason to change as things have been so stable. I know that Win7 is EOL with Microsoft, but I don't see that as much of an issue with the way I use it. I have no axe to grind with Win10, as I run it on other computers. For this particular application, Win7 seemed like a better choice at the time. Is my reasoning in error?

The main reason of uprgading Windows is compatibility with new versions of applications. An old version of GrandOrgue might work stable on Windows 7, but new versions of GrandOrgue are not tested against Windows 7 and may be not compatible.

**Reply by OrgelSpieler** [reply]
Reconfiguring with PA: ASIO provided stability.  I've had 16 channels running the RME ASIO driver for over 24 hours without so much as a hiccup.  I think it even sounds better; is that possible?

Thanks for being so patient with my question and timely responses.  It is greatly appreciated.

I guess the problem with the Windows drivers still exist, but it might be that I'm the only person using an RME AIO card where that problem shows up.  It might be due to the fact that the RME card looks like three interfaces to Windows and/or GrandOrgue (when using native Windows drivers).  Just a hunch.  What's odd is that it had been working.  With the ASIO driver it looks like one interface.

**Reply by oleg68** [reply]
> With the ASIO driver it looks like one interface.

Yes, I think this is the key. GrandOrgue works better with a single multi-channel audio interface than with several audio interfaces.

**Reply by OrgelSpieler** [reply]
Just a follow-up to say that after two weeks of daily use the RME ASIO driver continues to perform flawlessly with GO.  I'm even able to place the computer in "sleep" mode with GO active and re-awaken it and GO is ready.  Previously, when using the Windows drivers, I had to restart GO after the computer was in "sleep" mode.

**Comment by midimusic** [comment]
Have you considered that your problem may be caused by an intermittent  hardware fault? Possibly a bad connection somewhere and heat related. You could test this with a hair dryer - does the audio fail when you heat up various components in the computer? Memory cards often suffer with bad contacts even when new.

After GO audio has failed can you still play a recorded music track?

I would recommend going back to basics, removing all your sound cards and using the HD audio which is almost certainly built in to your computer. This should work for GO and all your other audio needs. Also I would never have anything to do with ASIO which is long outdated, has no benefits and often causes trouble.

csw900

**Reply by OrgelSpieler** [reply]
When I had the problem 18 mo ago, I considered a possible thermal cause, but there were too many things that didn't follow thermal failure behavior.  Now, I have two RME cards to test.  Both cards exhibit the same problem with GO since I updated.  However, if I use either card with a different music app on the same computer, I do not experience any failure, even after 48 hours of continuous use.  I can also rollback GO to the earlier version I had been using and the problem stops.  That tells me it's not the card, but some program glitch somewhere.  Thanks for the idea.

**Comment by midimusic** [comment]
Yes it does look like a GO problem - but remember that GO will be the only program which uses large parts of your memory and memory faults can produce some very peculiar symptoms. So I suggest you give your memory a thorough test before blaming GO itself.

csw900

**Reply by oleg68** [reply]
> Yes it does look like a GO problem

It looks like a **compatibility** problem.

> That tells me it's not the card, but some program glitch somewhere.

@Paul-Lau1 
We need to discover, is the problem related to your hardware and drivers or no. So please make the test with an internal audio with only 2 channels. Will sound stop or no? Otherwise we can not go further.
